client/state/timezones/reducer.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	TIMEZONES_RECEIVE,
+	TIMEZONES_REQUEST,
+	TIMEZONES_REQUEST_FAILURE,
+	TIMEZONES_REQUEST_SUCCESS
+} from 'state/action-types';
+import { timezonesSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+/**
+ * Collect and store timezones data
+ *
+ * @param  {Object} state - Current state
+ * @param  {Object} action - action payload
+ * @return {Object} updated state
+ */
+export const items = createReducer( {}, {
+	[ TIMEZONES_RECEIVE ]: ( state, {
+		rawOffsets,
+		timezonesByContinent,
+	} ) => ( { rawOffsets, timezonesByContinent } ),
+}, timezonesSchema );
+
+/**
+ * Track the timezones requesting process
```

Comment:
The main difference here is that `timezonesRequestAction` returns a simple object action, rather than a function thunk. This means that the data-layer hander can intercept the action and then decide on what to do. It isn't guaranteed that dispatching that action will trigger an API request.

Depending on our needs we might always want to fire an API request in the data-layer handler when we see an action, or throttle the same requests to every X seconds, or ignore them ( say if we're offline ).

Unless I'm misreading the wiring for`requestEligibility` @dmsnell it doesn't look like we're checking for in-flight requests in the data-layer. Did you or @aduth have any plans for utilities there? I assume we'll want to have common request strategies.